core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1031,20 +1033,22 @@ abstract class RDD[T: ClassTag](
     Array.concat(results: _*)
   }
 
+  def toLocalIterator : Iterator[T] = toLocalIterator(false)
+
   /**
    * Return an iterator that contains all of the elements in this RDD.
    *
    * The iterator will consume as much memory as the largest partition in this RDD.
+   * With prefetch it may consume up to the memory of the 2 largest partitions.
+   *
+   * @param prefetchPartitions If Spark should pre-fetch the next partition before it is needed.
    *
    * @note This results in multiple Spark jobs, and if the input RDD is the result
    * of a wide transformation (e.g. join with different partitioners), to avoid
    * recomputing the input RDD should be cached first.
    */
-  def toLocalIterator: Iterator[T] = withScope {
-    def collectPartition(p: Int): Array[T] = {
-      sc.runJob(this, (iter: Iterator[T]) => iter.toArray, Seq(p)).head
-    }
-    partitions.indices.iterator.flatMap(i => collectPartition(i))
+  def toLocalIterator(prefetchPartitions: Boolean = false): Iterator[T] = withScope {
```

Comment:
Should there be a default value since there is already `def toLocalIterator` above?

Reply:
the problem is the existing method is without ().
if i reused it and added default value, it'd bring source incompatibility